From 0f0ec540d11b347fc117ee4d1f8050358c332705 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Mon, 24 Mar 2025 01:10:03 +0000
Subject: [PATCH 6/7] i40e: remove gfp_mask from napi_alloc_skb()

This commit accommodates commit 6e9b01909a81("net: remove gfp_mask from
napi_alloc_skb()") which remove gfp_mask and change to napi_alloc_skb()
from __napi_alloc_skb.. This commit was merged in the v6.10 development
cycle and causes compilation failures with the v6.12 kernel, which are
listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=6e9b01909a81

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/i40e_txrx.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/i40e_txrx.c b/src/i40e_txrx.c
index e21ecf1..2dac2c2 100644
--- a/src/i40e_txrx.c
+++ b/src/i40e_txrx.c
@@ -2606,9 +2606,7 @@ static struct sk_buff *i40e_construct_skb(struct i40e_ring *rx_ring,
 #endif
 
 	/* allocate a skb to store the frags */
-	skb = __napi_alloc_skb(&rx_ring->q_vector->napi,
-			       I40E_RX_HDR_SIZE,
-			       GFP_ATOMIC | __GFP_NOWARN);
+	skb = napi_alloc_skb(&rx_ring->q_vector->napi, I40E_RX_HDR_SIZE);
 	if (unlikely(!skb))
 		return NULL;
 
-- 
2.47.1

